Delivery ancestor rapp and some bugfixes

stonier opened the pull request:
- ~~First iteration of delivery ancestor rapp (public interface/params just stubs)~~
- Bugfix to rocon app manager - was a broken assumption on current_rapp not being None
- Bugfix to rocon app utilities - moved variable use inside the scope of the try block before the excepts.
- Bugfix base name conversion so proper ros naming rules apply when not using uuids.
- Removed runtime information from the list_rapps topic (should all be in 'status').
- Syntax highlighting for the ugly/hard to read `rocon_app` command.

asmodehn committed:
removing running rapps from listrapps

stonier committed:
Merge pull request #289 from asmodehn/gocart

removing running rapps from listrapps

stonier committed:
[rocon_app_manager] bugfix handling of nonexistant current rapp.

Was mistakenly assuming that this is always available. This handles the
case when it is None.

stonier committed:
[rocon_app_utilities] pep8 fixes.

stonier committed:
[rocon_app_utilities] bugfix undefined variable call to ancestor_name.

Must be in scope before the exception branching is executed.

stonier committed:
[rocon_apps] first iteration of the delivery ancestor rapp.

stonier committed:
[rocon_app_manager] base name handling should be same for uuids or not.

Wasn't applying the correct conversion rules (lower, space->'_') for
standalone mode so that the namespaced topics/services conform to ros
conventions.

stonier committed:
[rocon_app_manager] remove debug print.

stonier committed:
[rocon_app_manager] bugfix the bugfix for status publishing

Didn't look around properly - this corrects the last bugfix to make sure
all variables are properly defined before publishing the status.

stonier committed:
[rocon_app_utilites] pep8 for rapp_cmd.py

stonier committed:
[rocon_app_utilities] syntax highlighting for rocon_app cmd output.

stonier commented on rocon_app_manager/src/rocon_app_manager/rapp.py:
```diff
@@ -29,6 +29,7 @@ class Rapp(object):
         Got many inspiration and imported from willow_app_manager implementation
     '''
     # I should add a __slots__ definition here to make it easy to read
+    # Dont think so : slots prevent using dict and other high level python stuff. (AlexV)
```
Aye, learned this after I wrote that comment a long time ago.

stonier committed:
[rocon_app_manager] remove unnecessary, and old, comment.

stonier committed:
[rocon_app_manager] pass a ros compatible name to the rapp.

stonier committed:
[rocon_app_manager] trivial

stonier committed:
[rocon_apps] next iteration of the delivery rapp interface.

jihoonl commented on rocon_apps/apps/delivery/delivery.interface:
```diff
@@ -0,0 +1,16 @@
+  - name: pose
+    type: geometry_msgs/PoseStamped
+  - name: robot_status
+    type: gopher_std_msgs/Status
```
where do I find this message format?

jihoonl commented:
All looks good except delivery rapp. I would like to see if this delivery rapp can be used for other robots like waiterbot and also gopher free.

Can you provide any pointer that describes how this interface is designed? and pointer to gopher_std_msgs repository?

stonier commented:
Oh, good point. Can't have this here if it's pointing to `gopher_std_msgs`.

The public interface is a work in progress. Looks like we have two options:
1. drop this rapp from here for now OR
2. move the supporting messages into rocon_msgs somewhere (more work but easier for collaborating on). 

What would you prefer?

jihoonl commented:
I would prefer to drop this rapp from this PR. and continue the rapp interface discussion in separate thread. How about this?

Having delivery meta rapp sounds good since we need for other service demos using waiterbot and turtlebot.

stonier committed:
[rocon_apps] delivery rapp moved out.

stonier commented:
Moved out the deliver rapp. Added you to the `gopher_concert` repo so you can see where this implementation is going and provide feedback on issues there.

[Delivery Rapp Thread](https://bitbucket.org/yujinrobot/gopher_concert/issue/1/delivery-rapp-specification).

jihoonl commented:
:+1:
